pkg/api/norman/store/scoped/store.go

```diff
@@ -25,7 +25,7 @@ func NewScopedStore(key string, store types.Store, pLister v3.ProjectLister) *St
 					return data, nil
 				}
 				v := convert.ToString(data[key])
-				if !strings.HasSuffix(v, ":"+convert.ToString(data[client.ProjectFieldNamespaceId])) && v != strings.Replace(convert.ToString(data[client.ProjectFieldNamespaceId]), "-", ":", 1) {
+				if !strings.HasSuffix(v, ":"+convert.ToString(data[client.ProjectFieldNamespaceId])) && strings.ReplaceAll(v, "-", ":") != strings.ReplaceAll(convert.ToString(data[client.ProjectFieldNamespaceId]), "-", ":") {
```

Comment:
What if instead of replacing `-` with `:` on both sides we just replace the `:` separator to `-` in the project id before comparing and make the whole conditional it a bit more succinct and easier to read?

```suggestion
				ns := convert.ToString(data[client.ProjectFieldNamespaceId])
				if !strings.HasSuffix(v, ":"+ns) && strings.Replace(v, ":", "-", 1) != ns {

```